bluepyemodel/icselector/modules/gene_selector.py

```diff
@@ -0,0 +1,229 @@
+        """Determine if a gene is present or not.
+
+        Args:
+            gene (DataFrame): single column from the gene table
+
+        Returns:
+            presence (array [bool]): gene is present
+        """
+
+        gene_present = gene["presence"]
+        if isinstance(gene_present, str):
+            presence = np.array([int(gene_present)], dtype=int) == 1
+        else:
+            presence = np.array(gene_present.values, dtype=int) == 1
+        return presence
+
+    @staticmethod
+    def _get_distributions(gene, presence, group_compartments):
+        """Get distribution per compartment for a given gene.
+
+        Args:
+            gene (DataFrame): single column from the gene table
+            presence (array [bool]): gene is present
+            group_compartments (bool): use 'all' if present in all compartments
+
+        Returns:
+            distros (Distribution): gene distribution per compartment
+        """
+
+        all_comps = None
+        dend = np.unique(np.array(gene.filter(regex="dend", axis=0).values[presence], dtype=str))
+        axon = np.unique(np.array(gene.filter(regex="axon", axis=0).values[presence], dtype=str))
+        soma = np.unique(np.array(gene.filter(regex="soma", axis=0).values[presence], dtype=str))
+
+        if len(dend) > 1:
+            logging.warning(
+                "Conflicting dendritic distributions for gene %s. Selected '%s' from %s",
+                gene.name,
+                dend[0],
+                dend,
+            )
+        dend = dend[0]
+        if len(axon) > 1:
+            logging.warning(
+                "Conflicting axonal distributions for gene %s. Selected '%s' from %s",
+                gene.name,
+                axon[0],
+                axon,
+            )
+        axon = axon[0]
+        if len(soma) > 1:
+            logging.warning(
+                "Conflicting somatic distributions for gene %s. Selected '%s' from %s",
+                gene.name,
+                soma[0],
+                soma,
+            )
+        soma = soma[0]
+
+        if (dend == axon) and (axon == soma):
+            all_comps = dend
+
+        def str_has_value(s):
+            return not ((s == "nan") or (s.strip() == ""))
+
+        distros = Distribution()
+        if all_comps and group_compartments:
+            if str_has_value(all_comps):
+                distros.all = all_comps
+        else:
+            distros.set_fields(somatic=soma, axonal=axon, basal=dend, apical=dend)
+        return distros
+
+    @staticmethod
+    def _get_gbar_max(gene):
+        """Get gbar_max for given gene.
+
+        Args:
+            gene (DataFrame): single column from the gene table
+
+        Returns:
+            gbar_max (float): gbar_max value
+        """
+
+        try:
+            gbar_max = gene["g_bar_max"]
+            gbar_max = np.array([float(g) for g in gbar_max.values])
+            # gbar_max = np.array(gbar_max.values, dtype = np.float)
```

Comment:
can remove this comment

Reply:
Done in [942a287](https://github.com/BlueBrain/BluePyEModel/pull/144/commits/942a287eb332cb58b5608511616eefcd2e76e880)